aocdbc/day_4_passport_processing.py

```diff
@@ -0,0 +1,107 @@
+from re import fullmatch
+import sys
+from typing import Dict, List, Optional
+
+from icontract import require, ensure
+
+_REQUIRED_KEYS = set(["byr", "iyr", "eyr", "hgt", "hcl", "ecl", "pid"])
+
+
+@ensure(lambda: True)
+def blank_line_split(text: str) -> List[str]:
+    # NOTE: I tried to do a lookahead for the 2nd newline, but split doesn't take
+    # a regex.
+    return text.split("\n\n")
+
+
+@ensure(lambda: True)
+def parse(passport: str) -> Optional[Dict[str, str]]:
```

Comment:
```suggestion
PASSPORT_RE = re.compile(
	r'^((byr|iyr|eyr|hgt|hcl|ecl|pid|cid):([^ ]+))'
	r'( (byr|iyr|eyr|hgt|hcl|ecl|pid|cid):([^ ]+))*\Z'
)
@require(lambda passport: PASSPORT_RE.match(passport))
def parse(passport: str) -> Optional[Dict[str, str]]:
```

Reply:
The problem describes these as "required fields," so I'm inclined to think that we would ideally accept other fields? I might just use `\w`. Changing.
Though, I have to say, this feels a little artificial - if I was going to write a regex to accept valid inputs, I might was well use it to extract them.

Reply:
You are correct, my comment is actually wrong -- you should use `\w` as you said.